cpl/module_block_data.F90

```diff
@@ -138,25 +139,62 @@ subroutine block_copy_1d_r8_to_2d_r8(destin_ptr, source_ptr, block, block_index,
     end if
 
     if (present(rc)) rc = localrc
-    
+
   end subroutine block_copy_1d_r8_to_2d_r8
 
+  subroutine block_copy_spval_1d_r8_to_2d_r8(destin_ptr, source_ptr, block, block_index, scale_factor, special_value, rc)
+
+    ! -- arguments
+    real(ESMF_KIND_R8),        pointer     :: destin_ptr(:,:)
+    real(kind=8),              pointer     :: source_ptr(:)
+    type(block_control_type),  intent(in)  :: block
+    integer,                   intent(in)  :: block_index
+    real(kind=8),              intent(in)  :: scale_factor
+    real(kind=8),              intent(in)  :: special_value
+    integer,         optional, intent(out) :: rc
+
+    ! -- local variables
+    integer      :: localrc
+    integer      :: i, ib, ix, j, jb
+
+    ! -- begin
+    localrc = ESMF_RC_PTR_NOTALLOC
+    if (associated(destin_ptr) .and. associated(source_ptr)) then
+       !$omp parallel do private(ix,ib,jb,i,j)
+       do ix = 1, block%blksz(block_index)
+          ib = block%index(block_index)%ii(ix)
+          jb = block%index(block_index)%jj(ix)
+          i = ib - block%isc + 1
+          j = jb - block%jsc + 1
+          if (source_ptr(ix) .ne. special_value) then
+             destin_ptr(i,j) = scale_factor * source_ptr(ix)
+          else
```

Comment:
lines 171 and 172 are not needed?

Reply:
I wasn't sure what value destin_ptr would have if it is not being filled with the source_ptr. I had originally 

```
j = hb - block%jsc + 1
destin_ptr(i,j) = special_value
if (source_ptr(ix) .ne. special_value) then
   destin_ptr(i,j) = .....
```

Reply:
I see.